lib/strings.nix

```diff
@@ -158,6 +158,24 @@ rec {
   */
   replicate = n: s: concatStrings (lib.lists.replicate n s);
 
+  /*
+    Remove leading and trailing whitespace from a string.
+
+    Whitespace is defined as any characters matching the regex
+    `[[:space:]]`.
+
+    Type: trim :: string -> string
+
+    Example:
+      trim "   hello   "
+      => "hello"
+  */
+  trim = s:
+    let
+      res = match "[[:space:]]*(.*[^[:space:]])[[:space:]]*" s;
+    in
+      optionalString (res != null) (head res);
```

Comment:
Can you explain how this works and why it's correct? :sweat_smile: Ideally as a code comment :)